ui/pages/connected-accounts/connected-accounts.component.js

```diff
@@ -1,75 +1,102 @@
 import PropTypes from 'prop-types';
-import React, { PureComponent } from 'react';
+import React from 'react';
+import { WALLET_SNAP_PERMISSION_KEY } from '@metamask/snaps-utils';
 import Popover from '../../components/ui/popover';
 import ConnectedAccountsList from '../../components/app/connected-accounts-list';
 import ConnectedAccountsPermissions from '../../components/app/connected-accounts-permissions';
 import { getURLHost } from '../../helpers/utils/util';
+import { useI18nContext } from '../../hooks/useI18nContext';
+import ConnectedSnaps from '../../components/app/connected-sites-list/connected-snaps';
+import { TextColor, TextVariant } from '../../helpers/constants/design-system';
+import { Box, Text } from '../../components/component-library';
 
-export default class ConnectedAccounts extends PureComponent {
-  static contextTypes = {
-    t: PropTypes.func.isRequired,
-  };
+export default function ConnectedAccounts({
+  accountToConnect = null,
+  activeTabOrigin,
+  isActiveTabExtension,
+  connectAccount,
+  connectedAccounts,
+  history,
+  mostRecentOverviewPage,
+  permissions = undefined,
+  selectedAddress,
+  removePermittedAccount,
+  setSelectedAddress,
+  subjectMetadata,
+  originOfActiveTab,
+  permissionSubjects,
+}) {
+  const t = useI18nContext();
+  const connectedSubjectsMetadata = subjectMetadata[originOfActiveTab];
+  const isPermissionSubject =
+    permissionSubjects[originOfActiveTab]?.origin ===
+      connectedSubjectsMetadata?.origin &&
+    permissionSubjects[originOfActiveTab]?.permissions[
+      WALLET_SNAP_PERMISSION_KEY
+    ];
 
-  static defaultProps = {
-    accountToConnect: null,
-    permissions: undefined,
-  };
+  const connectedPermissionSubjects =
```

Comment:
```suggestion
  const connectedSnaps =
```
Nit: perhaps this is more applicable?

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/20811/commits/d852bc33d66b72d794e1f5a236ecc06b4f5d8854